### PR TITLE
 log sessions

### LINKED ISSUE #592: Save python log in behavior folder
Right now the GUI saves a log each time it runs. We should save this python log in the saved data for each session. 
- The log is for every time the GUI opens, which could contain multiple sessions
- One option is to simply copy the log to the saved data folder. This would include all activity in the GUI before the session starts
- Another option is to write the logger to an additional file in the saved data folder when the session starts (or when the saved data folder is created), and then stop that additional logging when the session ends (although we want to log until the post-session weight is entered and saved)
- The log should be saved in the "win.TrainingFolder", which is the same folder the .json file is saved in. 
- The log should have be a .txt and should be named `python_gui_log.txt`

### PR DESCRIPTION
**Pull request by micahwoodard**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
- write log to session folder and close upon start of new session
### What issues or discussions does this update address?
- resolves #592 
### Describe the expected change in behavior from the perspective of the experimenter
- None
### Describe any manual update steps for task computers
- None
### Was this update tested in 446/447?
- [x] 446

### CODE REVIEW

**Comment by micahwoodard** [comment]
Tested locally but need to test on rig. Log of previous session will close users presses start, clear, new session, close, or load button after session. Note that if user declines a pop up box after pressing buttons, log file will not close.

File: src/foraging_gui/Foraging.py [hunk 1]
```diff
@@ -2263,13 +2266,17 @@ def closeEvent(self, event):
...
-                'Box {}, Foraging Close'.format(self.box_letter), 
+                'Box {}, Foraging Close'.format(self.box_letter),
                 'Close the GUI?',
-                QMessageBox.Yes | QMessageBox.No, QMessageBox.Yes)  
+                QMessageBox.Yes | QMessageBox.No, QMessageBox.Yes)
             if reply == QMessageBox.No:
                 event.ignore()
                 return
 
+        # if session log handler is not none, stop logging for previous session
```
Comment (alexpiet) [inline src/foraging_gui/Foraging.py:2276]: presumably this was part of testing and can be removed?

File: src/foraging_gui/Foraging.py [hunk 2]
```diff
@@ -3184,6 +3191,10 @@ def _Open(self,open_last = False,input_file = ''):
         self.load_tag=1
         self.ID.returnPressed.emit() # Mimic the return press event to auto-engage AutoTrain
 
+        # if session log handler is not none, stop logging for previous session
+        # if self.session_log_handler is not None:
+        #     self.end_session_log()
```
Comment (alexpiet) [inline src/foraging_gui/Foraging.py:3196]: again, can this commented code be removed?

File: src/foraging_gui/Foraging.py [hunk 3]
```diff
@@ -3248,28 +3259,31 @@ def _Clear(self):
...
         # If yes, clear parameters
         if reply == QMessageBox.Yes:
             for child in self.TrainingParameters.findChildren(QtWidgets.QLineEdit)+ self.centralwidget.findChildren(QtWidgets.QLineEdit):
                 if child.isEnabled():
                     child.clear()
+        else:
+            logging.info('Clearing declined')
+            return
+
+        # if session log handler is not none, stop logging for previous session
```
Comment (alexpiet) [inline src/foraging_gui/Foraging.py:3284]: again

**Review by alexpiet — APPROVED** [review]

**Pull request merged**